pkg/plugins/diff/service.go

```diff
@@ -0,0 +1,81 @@
+package diff
+
+import (
+	"context"
+	"errors"
+	"strconv"
+	"time"
+
+	"github.com/treeverse/lakefs/pkg/plugins"
+)
+
+type Differ interface {
+	Diff(context.Context, TablePaths, S3Creds) ([]*Diff, error)
+}
+
+var (
+	ErrUninitializedDiffService = errors.New("uninitialized diff service")
+)
+
+type TablePaths struct {
+	LeftTablePath  string
+	RightTablePath string
+	BaseTablePath  string
+}
+
+type S3Creds struct {
+	Key      string
+	Secret   string
+	Endpoint string
+}
+
+type Params struct {
+	TablePaths TablePaths
+	S3Creds    S3Creds
+}
+
+type Entry struct {
+	Version          string
+	Timestamp        time.Time
+	Operation        string
+	OperationContent map[string]string
```

Comment:
Please document these fields.

Also, can we make Operation a special type, with a closed list of known operations?